src/modules/job-exec/job-exec.c

```diff
@@ -1440,6 +1440,40 @@ static int unload_implementations (struct job_exec_ctx *ctx)
     return 0;
 }
 
+static void stats_cb (flux_t *h, flux_msg_handler_t *mh,
+                      const flux_msg_t *msg, void *arg)
+{
+    struct exec_implementation *impl;
+    json_t *o = NULL;
+    int i = 0;
+
+    if (!(o = json_object ())) {
+        errno = ENOMEM;
+        goto error;
+    }
+    while ((impl = implementations[i]) && impl->name) {
+        json_t *stats = NULL;
+        if (impl->stats) {
+            if ((*impl->stats) (&stats) == 0 && stats) {
+                if (json_object_set_new (o, impl->name, stats) < 0) {
+                    errno = ENOMEM;
+                    goto error;
+                }
+            }
+        }
+        i++;
+    }
```

Comment:
I thought only one implementation could be active at runtime?  (Am I confused?)

When I suggested a callback I just meant to integrate any general stats provided in the mainline with those produced by the active plugin.

Also (minor): new code should probably break long parameter lists at one per line, except in the case of "pack" functions and then it should keep keys and values together on a line

Reply:
> I thought only one implementation could be active at runtime? (Am I confused?)

Implementations are selected per job, e.g. if `attributes.system.exec.test.run_duration` is set then the test exec implementation is used.

Reply:
> Implementations are selected per job,

sorry my bad!

> By "mainline" you mean job-exec module?

Yes but my comment was misplaced due to an incorrect recollection of how stuff works here.

> Ahhh I think you're referring to the flux_respond_pack(),

Also the message handler function prototype.